pulsar-broker/src/test/java/org/apache/pulsar/client/impl/ReaderTest.java

```diff
@@ -397,4 +396,32 @@ public void testKeyHashRangeReader() throws IOException {
         }
 
     }
+
+    @Test
+    public void testReaderSubName() throws Exception {
+        final String topic = "persistent://my-property/my-ns/testReaderSubName";
+        final String subName = "my-sub-name";
+
+        Reader<String> reader = pulsarClient.newReader(Schema.STRING)
+                .subscriptionName(subName)
+                .topic(topic)
+                .startMessageId(MessageId.earliest)
+                .create();
+        ReaderImpl<String> readerImpl = (ReaderImpl<String>) reader;
+        assertEquals(readerImpl.getConsumer().getSubscription(), subName);
+        reader.close();
+
+        final String topic2 = "persistent://my-property/my-ns/testReaderSubName2";
+        admin.topics().createPartitionedTopic(topic2, 3);
+
+        reader = pulsarClient.newReader(Schema.STRING)
+                .subscriptionName(subName)
+                .topic(topic2)
+                .startMessageId(MessageId.earliest)
+                .create();
+        MultiTopicsReaderImpl<String> multiTopicsReader = (MultiTopicsReaderImpl<String>) reader;
+        multiTopicsReader.getMultiTopicsConsumer().getConsumers()
+                .forEach(consumerImpl -> assertEquals(consumerImpl.getSubscription(), subName));
+        multiTopicsReader.close();
```

Comment:
Overall looks good. I found if both the `subscriptionRolePrefix` and `subscriptionName` are specified, it will take the `subscriptionName`. Could you please also add a test case for this?